src/main/java/com/example/batteryprice/service/BatteryRepo.java

```diff
@@ -0,0 +1,24 @@
+package com.example.batteryprice.service;
+
+import com.example.batteryprice.model.Battery;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.stereotype.Service;
+
+@Service
+@Data
+@AllArgsConstructor
+public class BatteryRepo{
+
+    @Autowired
```

Comment:
You don't need this @Autowired here. Mark as NonNull and use required args constructor.